test/built-ins/object/entries/getter-adding-key.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2015 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Object.entries sees a new element added by a getter that is hit during iteration
+es7id: pending
+author: Jordan Harband
+---*/
+
+var bAddsC = {
+    a: 'A',
+    get b() {
+        this.c = 'C';
+        return 'B';
+    }
+};
+
+var result = Object.entries(bAddsC);
+
+assert.sameValue(Array.isArray(result), true, 'result is an array');
+assert.sameValue(result.length, 3, 'result has 3 items');
```

Comment:
The current spec has this note:
> [...] any property added or made enumerable during enumeration is not guaranteed to be included in the output;

But:
1. Notes are not normative.
2. The specification algorithm does not consider new properties. (O.\[[OwnPropertyKeys]\]() returns a fixed snapshot of the current property keys, so newly added properties are not part of the _ownKeys_ list.)